pkg/reconciler/pipelinerun/resources/pipelineref_test.go

```diff
@@ -654,6 +659,15 @@ func TestGetPipelineFunc_VerifyNoError(t *testing.T) {
 			if d := cmp.Diff(tc.expectedRefSource, source); d != "" {
 				t.Errorf("configSources did not match: %s", diff.PrintWantGot(d))
 			}
+			if tc.expectedVerificationResult == nil {
+				if vr != nil {
+					t.Errorf("VerificationResult did not match: want %v, got %v", tc.expectedVerificationResult, vr)
+				}
+				return
+			}
+			if tc.expectedVerificationResult.VerificationResultType != vr.VerificationResultType && errors.Is(vr.Err, tc.expectedVerificationResult.Err) {
+				t.Errorf("VerificationResult did not match: want %v, got %v", tc.expectedVerificationResult, vr)
+			}
 		})
```

Comment:
This part is a bit hard to understand. Instead of comparing all fields of `VerificationResult` individually, can we define a custom Comparer option for `VerificationResult` type and then just use `cmp.Diff`? See example https://github.com/google/go-cmp/issues/172#issuecomment-554799004.

This will clean up the comparison and also help when we want to introduce more subfields to `VerificationResult` in future.

Reply:
Thanks! Good suggestion !

Reply:
I think d3 is what we want and it works. 
https://go.dev/play/p/CXBSiXKlvW_r?v=goprev